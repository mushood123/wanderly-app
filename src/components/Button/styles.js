import styled from 'styled-components/native';

export const ButtonContainer = styled.TouchableOpacity`
  background-color: ${props => (props.$d ? '#A9A9A9' : '#007BFF')};
  padding-vertical: 12px;
  padding-horizontal: 25px;
  border-radius: 8px;
  align-items: center;
  justify-content: center;
`;

export const ButtonText = styled.Text`
  color: #ffffff;
  font-size: 16px;
  font-weight: bold;
`;
