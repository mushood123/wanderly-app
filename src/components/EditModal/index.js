import { EditModal } from './editModal';

export { EditModal };
