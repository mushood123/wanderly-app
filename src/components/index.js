import {FormField} from './FormField';
import {Text} from './Text';

export {FormField, Text};
