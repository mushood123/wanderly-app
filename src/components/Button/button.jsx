import React from 'react';
import {ButtonContainer, ButtonText} from './styles';

export const Button = ({
  onPress,
  title,
  style,
  textStyle,
  disabled = false,
}) => {
  return (
    <ButtonContainer
      onPress={onPress}
      activeOpacity={0.7}
      disabled={disabled}
      $d={disabled}
      style={style}>
      <ButtonText style={textStyle}>{title}</ButtonText>
    </ButtonContainer>
  );
};
