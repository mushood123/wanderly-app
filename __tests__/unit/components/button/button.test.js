import React from 'react';
import { Button } from '~src/components/Button';
import { render } from '../../utils/theme-wrapper/theme-wrapper';

describe('Button component', () => {
    it('matches the snapshot', () => {
        const { toJSON } = render(
            <Button
                onPress={() => { }}
                title="Test Button"
                style={{ backgroundColor: 'blue' }}
                textStyle={{ color: 'white' }}
            />
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('matches the snapshot when disabled', () => {
        const { toJSON } = render(
            <Button
                onPress={() => { }}
                title="Disabled Button"
                disabled
                style={{ backgroundColor: 'gray' }}
                textStyle={{ color: 'lightgray' }}
            />
        );
        expect(toJSON()).toMatchSnapshot();
    });
});
