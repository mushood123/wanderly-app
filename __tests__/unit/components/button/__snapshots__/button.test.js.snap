// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 25,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Test Button
  </Text>
</View>
`;

exports[`Button component matches the snapshot when disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "gray",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 25,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        {
          "color": "lightgray",
        },
      ]
    }
  >
    Disabled Button
  </Text>
</View>
`;
